test/routes/origins.js

```diff
@@ -1,10 +1,56 @@
 // Copyright (c) The Linux Foundation and others. Licensed under the MIT license.
 // SPDX-License-Identifier: MIT
 
-const { expect } = require('chai')
+const { expect, assert } = require('chai')
 const httpMocks = require('node-mocks-http')
 const sinon = require('sinon')
 const originCondaRoutes = require('../../routes/originConda')
+const proxyquire = require('proxyquire')
+const fs = require('fs')
+
+describe('Pypi origin routes', () => {
+  let router
+  let requestPromiseStub
+  const fixturePath = 'test/fixtures/origins/pypi'
+  beforeEach(() => {
+    requestPromiseStub = options => {
+      //Splitting url to extract the package name searched
+      const name = options.url.split('/')[4]
+      if (name === 'pand') throw { body: { message: 'Not Found' }, statusCode: 404 }
+      else if (name === 'pandas') return { body: loadFixture(`${fixturePath}/${name}.json`), statusCode: 200 }
+      // Return bad request error for all other input to check testcase with 400 status code error
+      throw { body: 'Bad Request', statusCode: 400 }
+    }
+    const Fetch = proxyquire('../../routes/originPyPi', { 'request-promise-native': requestPromiseStub })
+    router = Fetch(true)
+  })
+
+  afterEach(function () {
+    sinon.restore()
+  })
+
+  it('should return a valid response when a valid package is provided as input', async () => {
+    const response = await router._getPypiData('pandas')
+    expect(response.body.info.name).to.be.equal('pandas')
+  })
+  it('should return an empty response when a missing package is provided as input', async () => {
+    expect(await router._getPypiData('pand')).to.be.deep.equal([])
+  })
+  it('should return a valid error message when an error other than 404 occurs', async () => {
+    try {
+      await router._getPypiData('pan')
+      //Fail the test case if the above statement does not throw error
+      assert.fail()
```

Comment:
add a message to indicate the error, e.g. assert.fail('should have thrown error')